tests/algo/TestAlgo.cpp

```diff
@@ -259,3 +260,80 @@ TEST(Loads, base) {
     ASSERT_EQ(expected_loads[index].id, loads[index].id);
   }
 }
+
+TEST(Diagram, validDiagram) {
```

Comment:
The name of the suite should be consistent "Diagram" should be consistent with the rest of the file